std/format/package.d

```diff
@@ -45,61 +45,95 @@ $(TR $(TH Function Name) $(TH Description)
+More formally, $(I format strings) are composed according to the
+following grammar:
 
 $(PRE
 $(I FormatString):
-    $(I FormatStringItem)*
+    $(I FormatStringItem) $(I FormatString)
 $(I FormatStringItem):
-    $(B '%%')
-    $(B '%') $(I Position) $(I Flags) $(I Width) $(I Separator) $(I Precision) $(I FormatChar)
-    $(B '%$(LPAREN)') $(I FormatString) $(B '%$(RPAREN)')
-    $(B '%-$(LPAREN)') $(I FormatString) $(B '%$(RPAREN)')
-    $(I OtherCharacterExceptPercent)
+    $(I Character)
+    $(I FormatSpecifier)
+$(I FormatSpecifier):
+    $(B '%') $(I Parameters) $(I FormatIndicator)
+
+$(I FormatIndicator):
+    $(I FormatCharacter)
+    $(I CompoundIndicator)
+$(I FormatCharacter):
+    $(I see remark below)
+$(I CompoundIndicator):
+    $(B '$(LPAREN)') $(I FormatString) $(B '%$(RPAREN)')
+    $(B '$(LPAREN)') $(I FormatString) $(B '%|') $(I Delimiter) $(B '%$(RPAREN)')
+$(I Delimiter)
+    $(I empty)
+    $(I Character) $(I Delimiter)
+
+$(I Parameters):
+    $(I Position) $(I Flags) $(I Width) $(I Precision) $(I Separator)
 $(I Position):
     $(I empty)
     $(I Integer) $(B '$')
+    $(I Integer) $(B ':') $(I Integer) $(B '$')
+    $(I Integer) $(B ':') $(B '$')
 $(I Flags):
     $(I empty)
-    $(B '-') $(I Flags)
-    $(B '+') $(I Flags)
-    $(B '#') $(I Flags)
-    $(B '0') $(I Flags)
-    $(B ' ') $(I Flags)
+    $(I Flag) $(I Flags)
+$(I Flag):
+    $(B '-')|$(B '+')|$(B '&nbsp;')|$(B '0')|$(B '#')
 $(I Width):
+    $(I OptionalPositionalInteger)
+$(I Precision):
     $(I empty)
-    $(I Integer)
-    $(B '*')
+    $(B '.') $(I OptionalPositionalInteger)
 $(I Separator):
     $(I empty)
-    $(B ',')
-    $(B ',') $(B '?')
-    $(B ',') $(B '*') $(B '?')
-    $(B ',') $(I Integer) $(B '?')
-    $(B ',') $(B '*')
-    $(B ',') $(I Integer)
-$(I Precision):
+    $(B ',') $(I OptionalInteger)
+    $(B ',') $(I OptionalInteger) $(B '?')
+$(I OptionalInteger):
     $(I empty)
-    $(B '.')
-    $(B '.') $(I Integer)
-    $(B '.*')
+    $(I Integer)
+    $(B '*')
+$(I OptionalPositionalInteger):
+    $(I OptionalInteger)
+    $(B '*') $(I Integer) $(B '$')
+
+$(I Character)
+    $(B '%%')
+    $(I AnyCharacterExceptPercent)
 $(I Integer):
-    $(I Digit)
-    $(I Digit) $(I Integer)
+    $(I NonZeroDigit) $(I Digits)
+$(I Digits):
+    $(I empty)
+    $(I Digit) $(I Digits)
+$(I NonZeroDigit):
+    $(B '1')|$(B '2')|$(B '3')|$(B '4')|$(B '5')|$(B '6')|$(B '7')|$(B '8')|$(B '9')
 $(I Digit):
     $(B '0')|$(B '1')|$(B '2')|$(B '3')|$(B '4')|$(B '5')|$(B '6')|$(B '7')|$(B '8')|$(B '9')
-$(I FormatChar):
-    $(B 's')|$(B 'c')|$(B 'b')|$(B 'd')|$(B 'o')|$(B 'x')|$(B 'X')|$(B 'e')|$(B 'E')|$(B 'f')|$(B 'F')|$(B 'g')|$(B 'G')|$(B 'a')|$(B 'A')|$(B '|')
 )
 
+Note: $(I FormatCharacter) is left unspecified. Theoretically it can
+be any character, that has no other purpose in this grammer, but it
+is strongly recommended to stick to (lower- and uppercase) letters
+$(MDASH) this might even be enforced in the future.
```

Comment:
I don't think it's necessary to mention this.

Reply:
Maybe I wasn't clear, but I was just referring the last line (129) - I don't think it's necessary to mention that "this might even be enforced in the future". I'm fine with the rest of the paragraph.

Reply:
Oh, yeah, I got that wrong...